### ISSUE TITLE
clang-format don't report error but not format some files when set `Cpp11BracedListStyle: false`

### ISSUE THREAD

**Issue by yyjdelete** [llvm/llvm-project#63891]
tested with clang-format 16.0.6(latest stable), and 16.0.0 (https://github.com/llvm/llvm-project 08d094a0e457360ad8b94b017d2dc277e697ca76) which in bundled in vscode
No warn or error is reported by clang-format, but the struct init will not be formated
Set ColumnLimit to 0 like `BasedOnStyle: WebKit` does also works, but didn't found any thing about this in doc

### c file
```c
typedef struct
{
    int f1;
    int f2;
    int f3;
} _s2;

typedef struct
{
    int f1;
    _s2 f2;
} _s1;

_s1 s1 = {
    .f1 = 1,
    .f2 = {
        .f1 = 2,
                   .f2 = 2,           //it should be format
    },
};
```

### .clang-format config
```
---
BasedOnStyle: Microsoft
Language: Cpp

#ColumnLimit: 120
Cpp11BracedListStyle: false
```
